### ISSUE TITLE
[a11y][Tabs] Tab list does not have accurate tab count

### ISSUE THREAD

**Issue by chloerice** [Shopify/polaris-react#2442]
## Issue summary

Screen reader users are currently not receiving an accurate count of the number tabs with any browser + screen reader combination, including Firefox + NVDA and VoiceOver + iOS Safari/Safari/Chrome.

## Expected behavior

When using a screen reader, each tab reads as "1 of {TAB COUNT}". 

## Actual behavior

When using a screen reader, each tab reads as "1 of 1". 

## Steps to reproduce the problem

1. Enable VoiceOver in your Macbook system preferences.
2. Navigate the tabs with the Tab key then the arrow keys [in either example in the style guide](https://polaris.shopify.com/components/navigation/tabs)
3. Each tab will read as "1 of 1" as you navigate the tabs.

## Specifications

- *Are you using the React components?* (Y/N): Y
- *Polaris version number:* latest
- *Browser:* latest supported browsers
- *Device:* Macbook or iOS
- *Operating System:* Mojave

**Comment by dpersing** [comment]
Not all AT will read out the "X of Y" content, but for those that do we definitely want to make sure things get called out logically!

I think you're totally on the right track with the role structure, @chloerice! If I apply `role="presentation"` to the `<li>` elements, the "X of Y" numbers appear to read correctly.

It looks like the role was present before but was removed in August on in PR #1958 to address errors that were appearing in automated tests. However, it looks like the previous structure was a better fit for AT users.

**Comment by ginabak** [comment]
Closing in favour of #3647 🥳
